selfdrive/thermald/power_monitoring.py

```diff
@@ -59,92 +67,120 @@ def panda_current_to_actual_current(panda_current):
 
 
 class PowerMonitoring:
-  def __init__(self):
+  def __init__(self, params):
+    self.params = params
     self.last_measurement_time = None           # Used for integration delta
     self.power_used_uWh = 0                     # Integrated power usage in uWh since going into offroad
     self.next_pulsed_measurement_time = None
+    self.car_voltage_mV = 12e3                  # Low-passed version of health voltage
     self.integration_lock = threading.Lock()
 
+    car_battery_power_uWh = params.get("CarBatteryPower")
+    if car_battery_power_uWh is None:
+      # If unknown, we assume the car battery is almost dead
+      self.car_battery_power_uWh = (CAR_BATTERY_CAPACITY_uWh / 10)
+    else:
+      # Reset capacity if it's low
+      self.car_battery_power_uWh = max((CAR_BATTERY_CAPACITY_uWh / 10), int(car_battery_power_uWh))
+    
+
   # Calculation tick
   def calculate(self, health):
     try:
       now = sec_since_boot()
 
       # Check that time is valid
-      if datetime.datetime.fromtimestamp(now).year < 2019:
+      if datetime.datetime.now().year < 2019:
```

Comment:
Why do we care if the time is valid here? Do we need anything besides seconds since boot?

Reply:
Now that we switched to sec since boot, that should indeed be fine I guess.
That doesn't jump when the system time is updated I assume?

Reply:
Now that we switched to sec since boot, that should indeed be fine I guess.
That doesn't jump when the system time is updated I assume?

Reply:
Now that we switched to sec since boot, that should indeed be fine I guess.
That doesn't jump when the system time is updated I assume?

Reply:
No, it shouldn't

Reply:
Ok, removed!